### PR TITLE
feat: Holder of drop page iteration 

### LINKED ISSUE #8886: Holder of drop page iteration
> One item was used for a claim Chroma Lattice, and the second was not; they can not claim drop where it is required holderOf .motherboard as the current code uses the first one, which is claimed already for user.

was not a good design from me, lets fix it a bit. Better version hopefully coming with v2 drop page UI
- its clear when you hold more than 1 nft

# Design
- updated in figma https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=394-6126&mode=design&t=FmXDMzEHIN8lTS0n-11

image

- list of requirements, check or cross next to them

other possible states
image
- with the one failed, it shows the same as with 2 fails - requirements not met 

### available for mint
- this shows how much nfts from the collection you can still use to mint the drop 

image

there is also a tooltip: 
image

## Mobile 
just stacking things on top

image

thanks! lmk if anything 🚀

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Needs Design check

- @exezbcz please review

## Context

- [x] Closes #8886

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

connect wallet

CleanShot 2024-01-17 at 14 06 02@2x

requirements not met

CleanShot 2024-01-17 at 14 07 20@2x

ready to mint 

CleanShot 2024-01-17 at 14 05 24@2x

CleanShot 2024-01-18 at 14 40 16@2x

### CODE REVIEW

File: components/collection/drop/MintSection/MintRequirementItem.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return props.fulfilled ? 'check' : 'xmark'
+})
+
+const iconClass = computed(() => {
+  if (!accountId.value) {
+    return 'text-k-grey'
+  }
+
+  return props.fulfilled ? 'text-k-green' : 'text-k-red'
+})
```
Comment (Jarsen136) [inline components/collection/drop/MintSection/MintRequirementItem.vue:32]: I would unify these two computed value like
```
return {
   icon: 'lock-keyhole',
   class: 'text-k-grey'
}
```

**Review by Jarsen136 — COMMENTED** [review]
otherwise lgtm

**Review by Jarsen136 — APPROVED** [review]

**Review by preschian — APPROVED** [review]

**Comment by yangwao** [comment]
So this one solves if I own 2 motherboard items and 1 already been used, second will be used for the 2nd mint last one?

**Comment by hassnian** [comment]
> So this one solves if I own 2 motherboard items and 1 already been used, second will be used for the 2nd mint last one?

nope, from what I could test that **worked fine**

### Logic changes

calculate available nfts (not used)

CleanShot 2024-01-26 at 14 32 29@2x

 using the same logic , as this 

CleanShot 2024-01-26 at 14 34 26@2x

and based on that disable the button 

### UI changes

and the rest are ui changes , to fix this 
CleanShot 2024-01-26 at 14 38 23@2x

better ux 
CleanShot 2024-01-26 at 14 36 16@2x

**Comment by hassnian** [comment]
@yangwao also wanted to mention that I'm waiting for a **loading state** , the design didn't have it , already discussed that with @exezbcz on telegram.

**Comment by exezbcz** [comment]
@hassnian yup, something like this


updated in figma: 
image
https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=394-6126&mode=design&t=GucpYRCIobOYN8v5-11

**Comment by prury** [comment]
I have one account that had previously minted one chroma using a motherboard nft, then i asked Exez to lend me one of his motherboard NFTs to check if i could be eligible to mint another Chroma.

my account:
https://deploy-preview-9027--koda-canary.netlify.app/ahp/u/13s37n7Y9VvoF1JcqnQzbL2b76qb9vNQQyXvpvpProdrpvBe

NFT he lended to me: https://canary.kodadot.xyz/ahp/gallery/50-357

yet it does not identify that i have a second motherboard nft:

image

**Comment by hassnian** [comment]
> I have one account that had previously minted one chroma using a motherboard nft, then i asked Exez to lend me one of his motherboard NFTs to check if i could be eligible to mint another Chroma.
> 
> my account: https://deploy-preview-9027--koda-canary.netlify.app/ahp/u/13s37n7Y9VvoF1JcqnQzbL2b76qb9vNQQyXvpvpProdrpvBe
> 
> NFT he lended to me: https://canary.kodadot.xyz/ahp/gallery/50-357
> 
> yet it does not identify that i have a second motherboard nft:
> 
> image

@prury  from what I can see logic was ok, both nfts are claimed, as per the design `availabe for mint` count should be visible even after a drop is minted out (fixed mb) 

**btw you can use `/ahk/drops/flatwhite` to test**

check again

thanks

CleanShot 2024-01-30 at 09 58 59@2x

**Comment by yangwao** [comment]
image
let's merge it as it's getting old PR 


@vikiival has a point that would need in future to reflect new API with a new worker (which is not done yet)
- #9104

**Comment by vikiival** [comment]
It is non-blocking - https://github.com/kodadot/private-workers/issues/43

**Comment by yangwao** [comment]
Probably will go to another release or @hassnian if you can resolve conflicts under 30 min? :D

**Comment by hassnian** [comment]
> Probably will go to another release or @hassnian if you can resolve conflicts under 30 min? :D

on it

**Comment by yangwao** [comment]
love the energy

**Comment by yangwao** [comment]
Thanks!
pay 80 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $80 @ 6.64 USD/DOT ~ 12.048 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0x8b1da8b504aaf351c8fd30d1f7aac2ebd2ec635668256f984bc45674a55b0744](https://polkadot.subscan.io/extrinsic/0x8b1da8b504aaf351c8fd30d1f7aac2ebd2ec635668256f984bc45674a55b0744)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**